src/main/java/uk/gov/hmcts/reform/hmc/config/MessageProcessor.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.hmcts.reform.hmc.config;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.hmc.ApplicationParams;
+import uk.gov.hmcts.reform.hmc.client.futurehearing.ActiveDirectoryApiClient;
+import uk.gov.hmcts.reform.hmc.client.futurehearing.HearingManagementInterfaceApiClient;
+import uk.gov.hmcts.reform.hmc.repository.DefaultFutureHearingRepository;
+
+import java.util.Map;
+
+@Slf4j
+@Component
+public class MessageProcessor {
+
+    private final ApplicationParams applicationParams;
+    private final ActiveDirectoryApiClient activeDirectoryApiClient;
+    private final HearingManagementInterfaceApiClient hmiClient;
+
+    public MessageProcessor(ApplicationParams applicationParams,
+                            ActiveDirectoryApiClient activeDirectoryApiClient,
+                            HearingManagementInterfaceApiClient hmiClient) {
+        this.applicationParams = applicationParams;
+        this.activeDirectoryApiClient = activeDirectoryApiClient;
+        this.hmiClient = hmiClient;
+    }
+
+    public void processMessage(JsonNode message, MessageType messageType, Map<String, Object> applicationProperties) {
+        if (messageType != null) {
+
+            DefaultFutureHearingRepository defaultFutureHearingRepository =
+                new DefaultFutureHearingRepository(
+                    activeDirectoryApiClient,
+                    applicationParams,
+                    hmiClient
+                );
+
+            switch (messageType) {
+                case REQUEST_HEARING:
+                    log.info("Message of type REQUEST_HEARING received");
```

Comment:
All these log messages look like should be `debug` instead of `info`, so wouldn't want them as they are displaying constantly in logs unless there's an identifier for info?